internal/resources/network_peer.go

```diff
@@ -0,0 +1,418 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	api "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	network_peer_api "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api/network_peer"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &NetworkPeer{}
+	_ resource.ResourceWithConfigure   = &NetworkPeer{}
+	_ resource.ResourceWithImportState = &NetworkPeer{}
+)
+
+const errorMessageWhileNetworkPeerCreation = "There is an error during network peer creation. Please check in Capella to see if any hanging resources" +
+	" have been created, unexpected error: "
+
+type NetworkPeer struct {
+	*providerschema.Data
+}
+
+func NewNetworkPeer() resource.Resource {
+	return &NetworkPeer{}
+}
+
+func (n *NetworkPeer) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_network_peer"
+}
+
+func (n *NetworkPeer) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = NetworkPeerSchema()
+}
+
+// Create a network peer for the CSP.
+func (n *NetworkPeer) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.NetworkPeer
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := n.validateCreateNetworkPeer(plan)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error validating create network peering request",
+			"Could not validate network peering create request: "+err.Error(),
+		)
+		return
+	}
+
+	networkPeerRequest := network_peer_api.CreateNetworkPeeringRequest{
+		Name:         plan.Name.ValueString(),
+		ProviderType: plan.ProviderType.ValueString(),
+	}
+
+	//Check if AWSConfig or GCPConfig is not nil and marshal it
```

Comment:
Is this comment still relevant?

Reply:
no it was just for my reference, I've removed it now, thanks for pointing out 👍